wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1280,11 +1280,21 @@ public final class MLSService: MLSServiceInterface {
 
     }
 
-    /// Commit all pending proposals for all groups.
-    ///
-    /// - Throws: `MLSCommitPendingProposalsError` if proposals couldn't be commited.
+    public func commitPendingProposalsIfNeeded() {
+        guard let context = context else {
+            return
+        }
 
-    public func commitPendingProposals() async throws {
+        WaitingGroupTask(context: context) { [self] in
+            do {
+                try await commitPendingProposals()
+            } catch {
+                WireLogger.mls.error("Failed to commit pending proposals: \(String(describing: error))")
+            }
+        }
+    }
+
+    func commitPendingProposals() async throws {
```

Comment:
suggestion: add a comment stating that this may block for several seconds depending on the delay.